src/cli/bin.ts

```diff
@@ -95,19 +166,58 @@ yargs
       inputPath,
       outputPath,
       serversPath,
+      fromVersion,
+      toVersion,
       removeWidgets: keysOfWidgetPluginsToRemove,
       pivotInputPath,
       debug,
       stack,
     }) => {
+      const migrationFunctions = [["4.3", migrateUIFolder], ["5.0"]];
+      const versions = migrationFunctions.map(([version]) => version);
+      const fromVersions = versions.slice(0, -1);
+      const toVersions = versions.slice(1);
+      if (!fromVersions.includes(fromVersion)) {
+        throw new Error(`
+          You specified ${fromVersion} as the version to migrate your content from. It's not a valid version to migrate from.
+          The valid versions to migrate from are: ${fromVersions.join(", ")}.
+        `);
+      }
+      if (!toVersions.includes(toVersion)) {
+        throw new Error(`
+          You specified ${toVersion} as the version to migrate your content to. It's not a valid version to migrate to.
+          The valid versions to migrate to are: ${toVersions.join(", ")}.
+        `);
+      }
+      const fromVersionIndex = migrationFunctions.findIndex(
+        ([version]) => version === fromVersion,
+      );
+      const toVersionIndex = migrationFunctions.findIndex(
+        ([version]) => version === toVersion,
+      );
+      if (fromVersionIndex >= toVersionIndex) {
+        throw new Error(`
+          You specified ${fromVersion} as the version to migrate your content from, and ${toVersion} as the version to migrate your content to.
+          The version to migrate to must be higher than the version to migrate from.
+        `);
+      }
+      const functionsToCompose = migrationFunctions
+        .slice(fromVersionIndex, toVersionIndex)
+        // The second element of the sub-arrays of the filtered `migrationFunctions` are systematically migration functions.
+        .map(([, func]) => func) as MigrationFunction[];
+
       const legacyUIFolder = await fs.readJSON(inputPath);
       const legacyPivotFolder = pivotInputPath
         ? await fs.readJSON(pivotInputPath)
         : undefined;
       const servers = await fs.readJSON(serversPath);
 
-      const [migratedUIFolder, counters, errorReport] = await migrateUIFolder(
+      const [migratedUIFolder, counters, errorReport] = await compose(
```

Comment:
I'd find it clearer to add an intermediate step:
```suggestion
      const migrate = await compose(...functionsToCompose);
      const [migratedUIFolder, counters, errorReport] = await migrate(
```